File: tl-server/src/main/java/de/chiworks/tlserver/therapists/model/RequestStatus.java
package de.chiworks.tlserver.therapists.model;

import jakarta.annotation.Nonnull;

/**
 * Represents the status of a request to a therapist.
 * This enum defines various states that a request can be in.
 */
public enum RequestStatus {
    NOT_REQUESTED("NOT_REQUESTED"),
    REQUESTED("REQUESTED"),
    ACCEPTED("ACCEPTED"),
    REJECTED("REJECTED"),
    WAITLISTED("WAITLISTED"),
    WAITLIST_CLOSED_UNTIL("WAITLIST_CLOSED_UNTIL"),
    NOT_INTERESTED("NOT_INTERESTED");

    @Nonnull
    final String status;

    RequestStatus(@Nonnull String status) {
        this.status = status;
    }

    @Override
    @Nonnull
    public String toString() {
        return status;
    }
}

File: tl-webapp/projects/shared/src/lib/rest/model/RequestStatus.ts
export enum RequestStatus {
  NOT_REQUESTED = "NOT_REQUESTED",
  REQUESTED = "REQUESTED",
  WAITLISTED = "WAITLISTED",
  WAITLIST_CLOSED_UNTIL = "WAITLIST_CLOSED_UNTIL",
  ACCEPTED = "ACCEPTED",
  REJECTED = "REJECTED",
  NOT_INTERESTED = "NOT_INTERESTED",
}


export const RequestStatusOrder: Record<RequestStatus, number> = {
  [RequestStatus.ACCEPTED]: 0,
  [RequestStatus.NOT_REQUESTED]: 1,
  [RequestStatus.REQUESTED]: 2,
  [RequestStatus.WAITLISTED]: 3,
  [RequestStatus.WAITLIST_CLOSED_UNTIL]: 4,
  [RequestStatus.REJECTED]: 5,
  [RequestStatus.NOT_INTERESTED]: 6,
};
